integration-tests/jaxb/src/main/java/io/quarkus/it/jaxb/BookIBANField.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.it.jaxb;
+
+import jakarta.xml.bind.annotation.XmlAccessType;
+import jakarta.xml.bind.annotation.XmlAccessorType;
+import jakarta.xml.bind.annotation.XmlElement;
+import jakarta.xml.bind.annotation.XmlTransient;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlTransient
```

Comment:
Adding `@XmlTransient` here excludes BookIBANField from reflection registration, no? In that way we stop reproducing the original use case of https://github.com/quarkusio/quarkus/issues/36479 . Shouln't we better introduce a new class hierarchy for testing the exclusion of `@XmlTransient`?